src/SDK/Trace/Behavior/LoggerAwareTrait.php

```diff
@@ -23,20 +27,65 @@ public function setDefaultLogLevel(string $logLevel): void
         $this->defaultLogLevel = $logLevel;
     }
 
+    /**
+     * @return static
+     */
+    public function withLogger(LoggerInterface $logger)
```

Comment:
What's the purpose of this method? If you want to have a fluid interface you can simply override the `setLogger`method.

```php

    public function setLogger(LoggerInterface $logger): self
    {
        $this->logger = $logger;

        return $this;
    }
```

(I don't like fluid interface anyway, because it's semantically incorrect, but I know most people like it.)

Reply:
I've overridden `setLogger` for now (phan + psalm were unhappy). Whether the fluid interface gets to stay is up for debate, it just makes setting up via sdk a little easier.

Reply:
(I'm not sure that the fluid interface aids readability much, but it does allow you to still create a tracer in one go)

Reply:
Fluid interface is already used in many places in the library. Just because I don't like it, doesn't mean I need to have everything my way ;)